src/main/java/com/webdev/cosmo/cosmobackend/posts/config/PostConfig.java

```diff
@@ -0,0 +1,16 @@
+package com.webdev.cosmo.cosmobackend.posts.config;
+
+import com.webdev.cosmo.cosmobackend.posts.mapper.PostMapper;
+import com.webdev.cosmo.cosmobackend.posts.repository.PostRepository;
+import com.webdev.cosmo.cosmobackend.posts.service.PostServiceImpl;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class PostConfig {
+
+    @Bean
+    public PostServiceImpl postService(PostRepository repository, PostMapper mapper) {
+        return new PostServiceImpl(repository, mapper);
+    }
+}
```

Comment:
Instead of creating `PostConfig` with manual Bean declaration, you can just add `@Service` annotation to `PostServiceImpl` class. The `@Service` (or `@Component` for non-service classes) annotation will generate bean for you.

For example: [Osiris infrastructure/plant/](https://github.com/PatrykPaluch/osiris/tree/df16f7648df0726995f612636cd0607a5ea0fb57/osiris-plant-server/src/main/java/com/paluchp/apps/osiris/osiris_plant_server/infrastructure/plant) and [Osiris api/plant](https://github.com/PatrykPaluch/osiris/tree/df16f7648df0726995f612636cd0607a5ea0fb57/osiris-plant-server/src/main/java/com/paluchp/apps/osiris/osiris_plant_server/api/plant). As you can see, there is no `@Bean` methods, only `@Service`/`@Component`/`@Repositoty` annotations and constructors (generated by lombok). You don't even need `@Autowired` nor setters.

Reply:
As there are many schools I prefer approach with bean 😃